upgrade/v1.2.1.md

```diff
@@ -0,0 +1,45 @@
+# IRIS Hub Upgrade v1.2.1
+
+## Upgrade Proposal
+
+We are prepared to upgrade the IRIS Hub based upon
+
+- Version: `v1.2.1`
+- Commit hash: `6e1bdcaddfbd292d0be72be3682e65083381bfda`
+- Patch execute at Block Height `12636666`, please upgrade your nodes **BEFORE HEIGHT 12636666**
+
+## Upgrade Process
+
+Note: All nodes have to be upgraded **BEFORE HEIGHT 12636666**, whether it is a validator node, a seed node or a full node.
+
+The specific steps are as follows:
+
+```bash
+# 1. Compile and install the new version, go 1.16+ is required
+git clone https://github.com/irisnet/irishub
+cd irishub && git checkout v1.2.1
+make install
+
+# 2. Check the version number
+iris version --long
+name: iris
+server_name: iris
+version: 1.2.1
+commit: 65140224a90e294ed91df087de4d1d22916de282
+build_tags: netgo,ledger
+go: go version go1.16 darwin/amd64
+
+# 3. Close the old software
+kill -9 <iris-pid>
+
+# 4. Restart
+iris start --home=<path-to-your-home>
+```
+
+## After Patch Height
+
+After the patch height is reached:
+
+- If the Voting Power of the upgraded validator nodes is greater than 67%, the network will perform a logical switch to fix the irishub bug for executing ibc txs. Relayers can complete the previously pending IBC Txs. The unupgraded nodes need to upgrade the node version and restart with a snapshot;
+- If the Voting Power of the upgraded validator nodes is greater than 33% and less than 67%, the network will stop generating blocks. The unupgraded nodes are required to upgrade the node version, and restart with a snapshot that is less than the height of the patch.
+- If the Voting Power of the upgraded validator nodes is less than 33%, the network will continue to generate blocks, but the logic of bug fix will not be switched. The upgraded nodes need to roll back the node version and restart with a latest snapshot.
```

Comment:
If the Voting Power of all upgraded ... less than 33%, the patch upgrade will have failed but the network will continue generating blocks (without the bug fix).  Nodes already running v1.2.1 have to be rolled back to v1.2.0 and restarted with a snapshot generated after the patch height (snapshot URL will be published after the patch height is reached).